### DISCUSSION TITLE
What is the mutation key for?

### DISCUSSION THREAD

**Discussion by jennings** [TanStack/query#5787] [Q&A] [answered]
This feels like a dumb question, but what is useMutation's `mutationKey` for? I've looked through documentation and through discussions, but I can't seem to find an explanation for what it affects.

I assumed incorrectly that it would let multiple useMutation calls to share the mutation status and result.

From usage, it seems like it's at least used for lookup with useIsMutating, and I saw one discussion alluding to dehydrating and rehydrating paused mutations. Are these the only purposes?

**Comment by TkDodo** [comment, marked as answer]
it's an optional key to be able to set default options for certain mutations only. In v5, we have a new `useMutationState` hook where you can grab state of multiple mutations depending on key. Same for `useIsMutating`

**Reply by jennings** [reply]
I saw your answer and forgot to ever come back to thank you, so thank you!
